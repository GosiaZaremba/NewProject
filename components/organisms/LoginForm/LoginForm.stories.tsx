/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { LoginForm } from './LoginForm';
import { Colors } from '../../../constants/colors';
import { color } from '@storybook/addon-knobs';

storiesOf('organisms/LoginForm', module)
    .addDecorator((story) => story())
    .add('default', () => <LoginForm onSubmitForm={() => {}} />);
