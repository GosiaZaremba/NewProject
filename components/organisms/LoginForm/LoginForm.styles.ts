import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  outerContainer: {
    alignItems: 'center',
  },
  inputsContainer: {
    marginVertical: 5,
    width: '100%',
  },
  inputContainer: {
    marginTop: 20,
  },
  buttonContainer: {
    marginVertical: 15,
    width: '100%',
  },
  switch: {
    alignSelf: 'flex-start',
    flexDirection: 'row',
    alignItems: 'center',
  },
});
