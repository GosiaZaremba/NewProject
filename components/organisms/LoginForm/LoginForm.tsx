import React, { useRef, useState } from 'react';
import { View } from 'react-native';
import { Colors } from '../../../constants/colors';
import { CustomButton } from '../../atoms';
import {
    CustomInputLabel,
    CustomInputLabelReference,
    CustomSwitchLabel,
    CustomSwitchLabelReference,
} from '../../molecules';
import styles from './LoginForm.styles';

export type Props = {
    onSubmitForm: (
        email: string,
        password: string,
        rememberMe: boolean | undefined
    ) => void;
    testID?: string;
};

export const LoginForm: React.FC<Props> = ({ onSubmitForm, testID }) => {
    const emailInputReference = useRef<CustomInputLabelReference>(null);
    const passwordInputReference = useRef<CustomInputLabelReference>(null);
    const rememberMeSwitchReference = useRef<CustomSwitchLabelReference>(null);

    const onSubmit = () => {
        const emailValue = emailInputReference.current?.getValue() || '';
        const passwordValue = passwordInputReference.current?.getValue() || '';
        const rememberMeValue = rememberMeSwitchReference.current?.getValue();
        // console.log(
        //   'email',
        //   emailValue,
        //   'password',
        //   passwordValue,
        //   'remember',
        //   rememberMeValue,
        // );
        onSubmitForm(emailValue, passwordValue, rememberMeValue);
    };

    return (
        <View style={styles.outerContainer} testID={testID}>
            <View style={styles.inputsContainer}>
                <View style={styles.inputContainer}>
                    <CustomInputLabel
                        labelText={'Email'}
                        placeholder={'email'}
                        secureTextEntry={false}
                        ref={emailInputReference}
                    />
                </View>
                <View style={styles.inputContainer}>
                    <CustomInputLabel
                        labelText={'Password'}
                        placeholder={'password'}
                        secureTextEntry={true}
                        ref={passwordInputReference}
                    />
                </View>
            </View>
            <View style={styles.switch}>
                <CustomSwitchLabel
                    trackColorFalse={Colors.grey}
                    trackColorTrue={Colors.grey}
                    thumbColorOn={Colors.pink}
                    thumbColorOff={Colors.grey}
                    labelText={'Remember me?'}
                    ref={rememberMeSwitchReference}
                />
            </View>
            <View style={styles.buttonContainer}>
                <CustomButton
                    title={'Login'}
                    onPress={onSubmit}
                    color={Colors.pink}
                />
            </View>
        </View>
    );
};
