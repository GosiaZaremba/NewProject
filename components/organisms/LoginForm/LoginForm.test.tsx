import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import { LoginForm } from './LoginForm';

const mockOnPress = jest.fn();

describe('Organism LoginForm', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <LoginForm
                    onSubmitForm={mockOnPress}
                    testID={'organism-loginForm'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure submit button works correctly', () => {
        const { getByTestId } = render(
            <LoginForm
                onSubmitForm={mockOnPress}
                testID={'organism-loginForm'}
            />
        );

        const submitButton = getByTestId('organism-loginForm-submit-button');
        fireEvent.press(submitButton);
        expect(mockOnPress).toHaveBeenCalledTimes(1);
        expect(submitButton).toHaveTextContent('Login');
    });

    test('ensure email input works correctly', () => {
        const { getByTestId } = render(
            <LoginForm
                onSubmitForm={mockOnPress}
                testID={'organism-loginForm'}
            />
        );

        const inputEmail = getByTestId('organism-loginForm-email-input-input');
        expect(inputEmail.props.value).toBe('');
        fireEvent.changeText(inputEmail, 'mail@wp.pl');
        expect(inputEmail.props.value).toBe('mail@wp.pl');
    });

    test('ensure password input works correctly', () => {
        const { getByTestId } = render(
            <LoginForm
                onSubmitForm={mockOnPress}
                testID={'organism-loginForm'}
            />
        );

        const inputPassword = getByTestId(
            'organism-loginForm-password-input-input'
        );
        expect(inputPassword.props.value).toBe('');
        fireEvent.changeText(inputPassword, 'haslo');
        expect(inputPassword.props.value).toBe('haslo');
    });

    test('ensure switch works correctly', () => {
        const { getByTestId } = render(
            <LoginForm
                onSubmitForm={mockOnPress}
                testID={'organism-loginForm'}
            />
        );

        const loginSwitch = getByTestId(
            'organism-loginForm-login-switch-switch'
        );
        expect(loginSwitch.props.value).toBe(false);
        fireEvent(loginSwitch, 'onValueChange', { value: true });
        expect(loginSwitch.props.value).toBe(true);
    });

    test('ensure onSubmitForm works correctly', () => {
        const { getByTestId } = render(
            <LoginForm
                onSubmitForm={mockOnPress}
                testID={'organism-loginForm'}
            />
        );

        const loginSwitch = getByTestId(
            'organism-loginForm-login-switch-switch'
        );
        const inputPassword = getByTestId(
            'organism-loginForm-password-input-input'
        );
        const inputEmail = getByTestId('organism-loginForm-email-input-input');
        const submitButton = getByTestId('organism-loginForm-submit-button');
        fireEvent.changeText(inputPassword, 'haslo');
        fireEvent.changeText(inputEmail, 'mail@wp.pl');
        fireEvent(loginSwitch, 'onValueChange', { value: true });
        fireEvent.press(submitButton);
        expect(mockOnPress).toHaveBeenCalledWith('mail@wp.pl', 'haslo', true);
    });
});
