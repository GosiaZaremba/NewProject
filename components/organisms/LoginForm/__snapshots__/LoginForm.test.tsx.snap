// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism LoginForm Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "black",
              "marginBottom": 2,
            }
          }
          testID="organism-loginForm-email-label"
        >
          Email
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="email"
          secureTextEntry={false}
          style={
            Object {
              "borderColor": "#908D8E",
              "borderRadius": 5,
              "borderWidth": 1,
              "padding": 10,
              "width": "100%",
            }
          }
          testID="organism-loginForm-email-input"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "black",
              "marginBottom": 2,
            }
          }
          testID="organism-loginForm-password-label"
        >
          Password
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="password"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#908D8E",
              "borderRadius": 5,
              "borderWidth": 1,
              "padding": 10,
              "width": "100%",
            }
          }
          testID="organism-loginForm-password-input"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "marginTop": 10,
      }
    }
  >
    <View>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#908D8E"
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        testID="organism-loginForm-switch"
        thumbTintColor="#908D8E"
        tintColor="#908D8E"
        value={false}
      />
      <Text
        style={
          Object {
            "color": "black",
            "marginBottom": 2,
          }
        }
        testID="organism-loginForm-label"
      >
        Remember me?
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 15,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="Login"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="organism-loginForm-button"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#fc296f",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
